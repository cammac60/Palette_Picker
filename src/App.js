import React, { useState, useEffect } from 'react';
import { Route } from 'react-router-dom';
import Login from './components/Login/Login';
import Home from './components/Home/Home';
import { getProjects } from './apiCalls.js';
import './App.scss';

const App = () => {
  const [user, setUser] = useState('');
  const [projects, setProjects] = useState([]);
  const homeProps = {user, setUser, projects, setProjects};

  useEffect(() => {
    const fetchProjects = async () => {
      try {
        const response = await getProjects();
        setProjects(response.projects);
      } catch(error) {
          console.log(error);
      }
    };
    fetchProjects();
  });

  return (
    <div className="App">
      <Route exact path='/' render={() => <Login setUser={setUser}/>} />
      <Route exact path='/home' render={() => <Home {...homeProps}/>} />
    </div>
  )
}

export default App;
