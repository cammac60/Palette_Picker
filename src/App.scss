@import './variables.scss';

.App {
  background-color:
  height: 100vh;
  width: 100vw;
}
