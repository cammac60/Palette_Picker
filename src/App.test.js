import React from 'react';
import App from './App';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
// import { Route } from 'react-router-dom';

import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  it('Should match the snapshot', () => {
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Should render without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    , div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
