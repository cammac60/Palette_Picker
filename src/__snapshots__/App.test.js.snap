// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/home"
    render={[Function]}
  />
  <Route
    path="/projects/:id"
    render={[Function]}
  />
</div>
`;
